core/wrappers/errors.go

```diff
@@ -0,0 +1,32 @@
+package wrappers
+
+import (
+	"sync"
+)
+
+type Errs struct {
+	Err error
+	rw  *sync.RWMutex
+}
+
+func (errs *Errs) Errored() bool {
+	errs.rw.RLock()
+	defer errs.rw.RUnlock()
+	return errs.Err != nil
+}
+
+func (errs *Errs) Add(errors ...error) {
+	errs.rw.RLock()
+	defer errs.rw.RUnlock()
+	if errs.Err == nil {
+		for _, err := range errors {
+			if err != nil {
+				errs.rw.RUnlock()
```

Comment:
I believe this will cause a panic if an error is ever added since we have called `RUnlock` here without re-acquiring the read lock before the deferred RUnlock will execute.